server/controllers/comments.js

```diff
@@ -44,43 +86,109 @@ const addCommentToPost = async (req, res) => {
   }
 };
 
+// @route   POST '/api/posts/:postId/comments/:commentId/replies'
+// @desc    add a reply to a specific comment
+// @access  private
+const addReplyToComment = async (req, res) => {
+  try {
+    const post = await getPost(req.params.postId);
+    const comment = await getComment(req.params.commentId);
+
+    if (!comment) {
+      return serverResponse(res, 404, { message: "Comment doesn't exist" });
+    }
+
+    const reply = {
+      ...req.body,
+      user: req.user.sub,
+      post: req.params.postId,
+      replyToComment: comment.replyToComment || req.params.commentId
+    };
+
+    post.comments++;
+    await post.save();
+
+    const response = await addComment(reply);
+
+    return serverResponse(res, 200, response);
+  } catch (error) {
+    return serverResponse(res, 500, {
+      message: 'Internal error while trying to add a reply'
+    });
+  }
+};
+
 // @route   DELETE '/api/posts/:postId/comments/:commentId'
-// @desc    Remove a single comment from a post
+// @desc    Remove a single comment/reply
 // @access  private
-const removeCommentFromPost = async (req, res) => {
+const removeAComment = async (req, res) => {
   try {
+    const getAComment = await getComment(req.params.commentId);
+    const replies = await getRepliesOfComment(req.params.commentId);
+
+    if (!getAComment) {
+      return serverResponse(res, 404, { message: "Comment doesn't exist" });
+    }
+
+    if (!requesterIsAuthenticatedUser(req.user.sub, getAComment.user)) {
+      return serverResponse(res, 400, {
+        message: 'Unauthorized!'
+      });
+    }
+
+    await removeLikesFromComment(req.params.commentId);
+    replies.forEach(async reply => {
+      await removeLikesFromComment(reply._id);
+    });
+    await removeAllCommentReplies(req.params.commentId);
```

Comment:
This can be made faster also

```
const removeLikesFromReplyPromises = replies.map(reply => removeLikesFromComment(reply._id));

await Promise.all([
  removeLikesFromComment(req.params.commentId),
  removeAllCommentReplies(req.params.commentId),
  ...removeLikesFromReplyPromises
])
```